src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -678,6 +680,10 @@ public void getDocContent(Request req, Response resp) throws IOException {
       // Generate Acl if aclSql is provided.
       if (aclSql != null) {
         resp.setAcl(getAcl(conn, id.getUniqueId()));
```

Comment:
Perhaps rename getAcl to getAclFromAclSql (to disambiguate that it is not running the content sql)?

Reply:
This call site for getAcl isn't running the content SQL, but when called from isUserAuthorized it is. I don't think that exposing the conditional from getAclFromDb in isUserAuthorized feels right. Perhaps better would be to bury this check for aclSql != null in getAcl as well, so that both call sites just always call getAcl, which figures out how to do the work.